csharp/src/Ice/CollocatedRequestHandler.cs

```diff
@@ -151,9 +151,17 @@ private async ValueTask InvokeAllAsync(OutgoingRequestFrame outgoingRequest, int
             {
                 if (_adapter.Communicator.TraceLevels.Protocol >= 1)
                 {
-                    // TODO we need a better API for tracing
-                    List<ArraySegment<byte>> requestData = Ice1Definitions.GetRequestData(outgoingRequest, requestId);
-                    TraceUtil.TraceSend(_adapter.Communicator, requestData);
+                    TraceFrame(_adapter.Communicator, requestId, outgoingRequest);
```

Comment:
I would simply add a second TraceCollocatedFrame version that takes the request ID instead of having the code here.